packages/reports/reports.hjson

```diff
@@ -0,0 +1,212 @@
+          }
+      ]
+    }
+  },
+  {
+    "id": "TO_RH_Descriptive_FP01_02",
+    "config": {
+      "fetch": [
+        {
+          "fetchAnalytics": {
+            "dataElementCodes": [
+              "Family_Planning_Acceptors_Other",
+              "Family_Planning_Acceptors_Condom_Female",
+              "Family_Planning_Acceptors_Natural_Method",
+              "Family_Planning_Acceptors_Depo_Provera",
+              "Family_Planning_Acceptors_Jadelle",
+              "Family_Planning_Acceptors_Pill_Single",
+              "Family_Planning_Acceptors_Sterilization_Male",
+              "Family_Planning_Acceptors_Sterilization_Female",
+              "Family_Planning_Acceptors_Pill_Combined",
+              "Family_Planning_Acceptors_Condom_Male",
+              "Family_Planning_Acceptors_IUD",
+              "POP21AND23",
+              "POP15",
+              "POP17AND19",
+              "POP13",
+              "POP11"
+            ],
+            "aggregationType": "RAW",
+            "dataServices": [
+              {
+                "isDataRegional": false
+              }
+            ]
+          }
+        }
+      ],
+      "transform": [
+        {
+            "transform": "select",
+            "$.dataElement": "$.value",
+            "'period'": "$.period"
+        },
+        {
+          "transform": "aggregate",
+          "period": "group",
+          "...": "sum"
+        },
+        // {
+        //   "transform": "select",
+        //   "'Contraceptive Prevalence Rate'": "100 * add([$.Family_Planning_Acceptors_Other, $.Family_Planning_Acceptors_Condom_Female, $.Family_Planning_Acceptors_Natural_Method, $.Family_Planning_Acceptors_Depo_Provera, $.Family_Planning_Acceptors_Jadelle, $.Family_Planning_Acceptors_Pill_Single, $.Family_Planning_Acceptors_Sterilization_Male, $.Family_Planning_Acceptors_Sterilization_Female, $.Family_Planning_Acceptors_Pill_Combined, $.Family_Planning_Acceptors_Condom_Male, $.Family_Planning_Acceptors_IUD]) / add([$.POP21AND23, $.POP15, $.POP17AND19, $.POP13, $.POP11])",
+        //   "'period'": "$.period"
+        // }
+      ],
+      "output": []
+    }
+  },
+  {
+    "id": "UNFPA_Region_Facilities_Offering_Services_At_Least_1_Family_Planning",
+    "config": {
+      "fetch": [
+        {
+          "fetchAnalytics": {
+            "dataElementCodes": [
+              "RHS4UNFPA809",
+              "RHS1UNFPA03"
+            ],
+            "aggregationType": "RAW",
+            "dataServices": [
+              {
+                "isDataRegional": true
+              }
+            ]
+          }
+        }
+      ],
+      "transform": [
+        {
+            "transform": "select",
+            "$.dataElement": "$.value",
+            "'period'": "convertToPeriod($.period, 'QUARTER')",
+            "'organisationUnit'": "$.organisationUnit"
+        },
+        {
+            "transform": "filter",
+            "where": "not (exists($.RHS4UNFPA809) and gt(1, $.RHS4UNFPA809))"
```

Comment:
oh cool...!

Reply:
Yeah! It does boolean logic by itself, and we can even use our custom defined functions with it quite gracefully 😄

I've been using `mathjs` so far just to get things running, but I'm kinda tempted to use it for the core expression parsing for this config given how well it's been working. Not definite in that stance though, and I think it requires us to consider a few things, but even if we don't I think using a similar design for our own expression parsing libraries could be really neat